### PR TITLE
Implement data access module, replace file resource

### PR DESCRIPTION
**Pull request by nezhar**
This is the first step into movidng from [ngx-resource-factory](https://github.com/beachmachine/ngx-resource-factory) to our [rest api client](https://www.npmjs.com/package/@snypy/rest-client) generate from the OpenAPI document

This PR installs and configures the module, prepares the environment for paralel usage of services and resources and replaces FileResource with FileService.

### CODE REVIEW

File: src/app/components/snippet/snippet.component.ts [hunk 1]
```diff
@@ -31,21 +32,13 @@ export class SnippetComponent implements OnInit, OnDestroy {
...
-          .$promise.then(data => {
-            this.files = data;
-          })
-          .catch(error => {
-            console.log(error);
-          });
+    this.availableLabelsSubscription = this.activeSnippet$.subscribe(activeSnippet => {
+      if (activeSnippet) {
+        this.activeSnippet = activeSnippet;
+        this.files$ = this.fileService.fileList(null, null, activeSnippet.pk);
```
Comment (nezhar) [inline src/app/components/snippet/snippet.component.ts:41]: @sterapps do you know if we can use keyword arguments here?

Comment (nezhar) [reply src/app/components/snippet/snippet.component.ts:41]: This is a bit strange in the [documentation](https://openapi-generator.tech/docs/generators/typescript-angular/), but it seems that `useSingleRequestParameter=true` will do the job for us

Comment (sterapps) [reply src/app/components/snippet/snippet.component.ts:41]: Yes useSingleRequestParameter=true does the trick 👍🏼

Comment (nezhar) [reply src/app/components/snippet/snippet.component.ts:41]: I will create `1.0.2` as a new release where this option is enabled

File: src/app/components/snippet/snippet.component.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
 import { SelectSnapshot } from '@ngxs-labs/select-snapshot';
 import { Select } from '@ngxs/store';
+import { FileService } from '@snypy/rest-client';
 import { ResourceModel } from 'ngx-resource-factory/resource/resource-model';
```
Comment (sterapps) [inline src/app/components/snippet/snippet.component.ts:5]: Please remove unused imports.

Comment (nezhar) [reply src/app/components/snippet/snippet.component.ts]: Fixed. I also refactored some logic regarding the label loading.

**Review by sterapps — APPROVED** [review]

**Pull request merged**